frontend/public/components/monitoring/query-browser.tsx

```diff
@@ -317,13 +326,18 @@ const QueryBrowser_: React.FC<QueryBrowserProps> = ({
   // For the default time span, use the first of the suggested span options that is at least as long as defaultTimespan
   const defaultSpanText = spans.find((s) => parsePrometheusDuration(s) >= defaultTimespan);
 
+  const [span, setSpan] = React.useState(parsePrometheusDuration(defaultSpanText));
+
+  // Limit the number of samples so that the step size doesn't fall below minStep
+  const minSamplesForSpan = Math.min(span / minStep, minSamples);
```

Comment:
Do we need a `Math.floor` in case this isn't an even number?

Reply:
Should this be `Math.max`?

Reply:
Thanks @spadgett. After thinking about this a bit more, I've simplified it to just use `minSamples` instead of defining `minSamplesForSpan`, which I think is a better approach.

Also added `Math.round` to avoid non integers.